firmware/main.c

```diff
@@ -3,6 +3,9 @@
 PortData port_a;
 PortData port_b;
 
+// TODO: flash buffer could be shared with the port, because they are not used simultaneously
+USB_ALIGN u8 flash_buffer[FLASH_BUFFER_SIZE];
```

Comment:
SAMD21 USB controller requires it. I assume the memory is 32-bits wide internally and the DMA in the USB controller accesses in 32-bit chunks and doesn't implement the hardware to do unaligned accesses split across words.